### ISSUE TITLE
Plots now include rates for systems with VariableRateJumps

### ISSUE THREAD

**Issue by isaacsas** [SciML/JumpProcesses.jl#436]
Sometime after 9.10 the plot recipe has started to include propensity variables within plots, i.e. compare 

https://docs.sciml.ai/JumpProcesses/v9.12/tutorials/discrete_stochastic_example/c355f569.svg

vs. the 9.10.1 version:

https://docs.sciml.ai/JumpProcesses/v9.10/tutorials/discrete_stochastic_example/86922c19.svg

I assume this is a plot-recipe change in SciMLBase, but don't know how / why it was ignoring them before.

**Comment by ChrisRackauckas** [comment]
I didn't know it dropped some before. The current version is what we'd considered the correct one though?

**Comment by isaacsas** [comment]
I thought the goal was to handle the ExtendedJumpArray aspect from users and make solutions look as much like ODE solutions as possible?

**Comment by ChrisRackauckas** [comment]
oh true. Yeah I see, that's why they were hidden 😅. I'm not sure why that would have changed with a release. That's pretty hard to test too. I may have to dig in.

**Comment by isaacsas** [comment]
FWIW, the last version that didn't have this behavior was on SciMLBase 2.10, while the first version exhibiting it is on 2.31. For JumpProcesses the change occurred with 9.11 but on quick glance I see nothing in the merged PRs that could cause this within JumpProcesses.

**Comment by isaacsas** [comment]
OK, I've verified the issue appears with SciMLBase 2.12 which had only this PR:

https://github.com/SciML/SciMLBase.jl/pull/572

which updated the plot recipe to use SII. So it seems that changed the indexing in a way that the propensity values are now being plotted. 

Here is a MWE if it would be useful for testing:
```julia
 using JumpProcesses, OrdinaryDiffEq, Plots
rate(u,p,t) = (1+t)*u[1]
affect(integrator) = (integrator.u[1] -= 1; nothing)
vrj = VariableRateJump(rate,affect)
oprob = ODEProblem((du,u,p,t) -> (du .=0; nothing), [10.0], (0.0, 10.0))
jprob = JumpProblem(oprob, Direct(), vrj)
sol = solve(jprob, Tsit5())
plot(sol)
```
Should I move this issue to SciMLBase?

**Comment by ChrisRackauckas** [comment]
Yeah. The plot recipe does not have a real interface to opt variables in/out by default. It worked mostly because of a quirk of indexing and size checking that we probably shouldn't be relying on. It needs to be made into a real documented and tested interface.
